flink-external-resources/flink-external-resource-gpu/src/test/resources/testing-gpu-discovery.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+source "$(dirname "$0")"/../../main/resources/gpu-discovery-util.sh
+
+if [ $# -lt 1 ]; then
+  echo "Usage: ./testing-gpu-discovery.sh gpu-amount [--privilege] [--check-dead] [--assign-file filePath] [exit-non-zero]"
+  exit 1
+fi
+
+AMOUNT=$1
+shift
+ASSIGNMENT_FILE="/var/tmp/flink-gpu-assignment"
+PRIVILEGE_OPTION=""
+CHECK_DEAD_PROCESS=""
+EXIT_NON_ZERO=""
+
+while [[ $# -ge 1 ]]
+do
+key="$1"
+  case $key in
+    --privilege)
+    PRIVILEGE_OPTION="privilege"
+    ;;
+    --check-dead)
+    CHECK_DEAD_PROCESS="check"
+    ;;
+    --assign-file)
+    ASSIGNMENT_FILE="$2"
+    shift
+    ;;
+    --exit-non-zero)
+    EXIT_NON_ZERO="exit-non-zero"
+    ;;
+    *)
+    # unknown option
+    ;;
+  esac
+  shift
+done
+
+if [ $AMOUNT -eq 0 ]; then
+  exit 0
+fi
+
+if [ "$EXIT_NON_ZERO" == "exit-non-zero" ]; then
+  exit 1
+fi
+
+indexes=($(seq 0 1 3))
```

Comment:
I would suggest to either make the total available gpu amount defined by argument, or document in the beginning that this script always pretend having 4 available gpus.
I personally prefer the argument approach.